### ISSUE TITLE
Dependencies not getting installed and being removed on update

### ISSUE THREAD

**Issue by Vaintti** [jorgebucaran/fisher#579]
Sometimes when I try to install a package with dependencies the dependencies don't get installed. Also when running fisher to update the packages it removes the dependencies. Happens to me with this package https://github.com/franciscolourenco/done Running the fisher add couple of times installs the dependencies (fishpkg/fish-humanize-duration) but doesn't work properly the first time.

Environment:
fish version 3.1.0
fisher version 3.2.11
macOS Catalina 10.15.5

**Comment by jorgebucaran** [comment]
@Vaintti Sorry, but I can't reproduce this. Everything works as expected.

<img width="1440" alt="スクリーンショット 2020-07-24 18 06 16" src="https://user-images.githubusercontent.com/56996/88376441-5fac9f80-cdd8-11ea-8689-05c461b07260.png">

**Comment by jorgebucaran** [comment]
@Vaintti Did you expect fishpkg/fish-humanize-duration to be added to your fishfile? Because that's not what fisher does. Only `franciscolourenco/done` is.

**Comment by Vaintti** [comment]
In your case it tells you it installed two packages but for me it was only one. Also I got errors about the missing humanize package. You could try to remove and install again to see if that does it. Running fisher to update packages always/almost always removes the dependency and breaks it. It  works for me 10% of the time. Don't know what's up with that.

**Comment by jorgebucaran** [comment]
@Vaintti Let's try this:

- Remove your `config.fish` and try again, if the problem persists:
  - Post your `config.fish`, please edit out any sensitive information.
  - Share a screenshot just like above with the problem you are having.
  - Show me this: `la ~/.local/share/ ~/.config/fisher/ ~/.config/fish`.

**Comment by jorgebucaran** [comment]
@Vaintti Any updates on this?

**Comment by jorgebucaran** [comment]
@Vaintti Any chance you can look at my request?

**Comment by Vaintti** [comment]
Yeah sorry I don't really use this computer with the problem outside of work hours. Removing config.fish didn't really help 
![Screenshot 2020-07-27 at 13 23 30](https://user-images.githubusercontent.com/11636057/88532034-89e6a300-d00c-11ea-9057-4e9aca2c17d1.png)

My config.fish only had two aliases and now nothing since I moved it away for the test.

```
ls: /Users/anttivainikka/.config/fisher/: No such file or directory
/Users/anttivainikka/.config/fish:
total 16
drwx------   7 anttivainikka  staff   224B Jul 27 13:21 ./
drwx------   5 anttivainikka  staff   160B Jul 16 11:22 ../
drwxr-xr-x   3 anttivainikka  staff    96B Jul 16 11:07 completions/
drwxr-xr-x   4 anttivainikka  staff   128B Jul 27 13:22 conf.d/
-rw-r--r--   1 anttivainikka  staff   2.9K Mar  2 13:59 fish_variables
-rw-r--r--   1 anttivainikka  staff    58B Jul 27 13:22 fishfile
drwxr-xr-x  31 anttivainikka  staff   992B Jul 27 13:22 functions/

/Users/anttivainikka/.local/share/:
total 8
drwx------   6 anttivainikka  staff   192B Jul 27 13:22 ./
drwx------   3 anttivainikka  staff    96B Oct 28  2019 ../
drwx------   4 anttivainikka  staff   128B Jul 24 11:37 fish/
drwxr-xr-x   4 anttivainikka  staff   128B Jul 27 13:22 fisher/
drwxr-xr-x  22 anttivainikka  staff   704B Mar  2 14:00 omf/
-rw-------   1 anttivainikka  staff   856B May  6 13:08 recently-used.xbel
```

**Comment by Vaintti** [comment]
fisher ls here too

edc/bass
fishpkg/fish-prompt-metro
franciscolourenco/done

**Comment by jorgebucaran** [comment]
@Vaintti Could you also show me what's in your `fishfile`?

**Comment by jorgebucaran** [comment]
Have you installed something that modifies the default behavior of `cp`, `mv`, etc? Or have you messed with file permissions in your filesystem, specifically `~/.config/*`?

**Comment by Vaintti** [comment]
I haven't installed anything that should modify cp or mv and I don't recall messing with permissions in my filesystem

```
 ~  cat ~/.config/fish/fishfile
fishpkg/fish-prompt-metro
edc/bass
franciscolourenco/done
```

```
 ~  printenv $XDG_CACHE_HOME
COLORFGBG=15;0
COLORTERM=truecolor
HOME=/Users/anttivainikka
ITERM_PROFILE=Default
ITERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
LC_CTYPE=UTF-8
LC_TERMINAL=iTerm2
LC_TERMINAL_VERSION=3.3.12
LOGNAME=anttivainikka
OMF_CONFIG=/Users/anttivainikka/.config/omf
OMF_PATH=/Users/anttivainikka/.local/share/omf
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
PWD=/Users/anttivainikka
SHELL=/usr/local/bin/fish
SHLVL=1
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.QY4EnwoKcA/Listeners
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.3.12
TERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
TMPDIR=/var/folders/s6/16dyg8sd01d3b1xp4qwrx_h80000gn/T/
USER=anttivainikka
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
```

```
 ~  printenv $XDG_CONFIG_HOME
COLORFGBG=15;0
COLORTERM=truecolor
HOME=/Users/anttivainikka
ITERM_PROFILE=Default
ITERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
LC_CTYPE=UTF-8
LC_TERMINAL=iTerm2
LC_TERMINAL_VERSION=3.3.12
LOGNAME=anttivainikka
OMF_CONFIG=/Users/anttivainikka/.config/omf
OMF_PATH=/Users/anttivainikka/.local/share/omf
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
PWD=/Users/anttivainikka
SHELL=/usr/local/bin/fish
SHLVL=1
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.QY4EnwoKcA/Listeners
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.3.12
TERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
TMPDIR=/var/folders/s6/16dyg8sd01d3b1xp4qwrx_h80000gn/T/
USER=anttivainikka
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
```

```
 ~  printenv $XDG_DATA_HOME
COLORFGBG=15;0
COLORTERM=truecolor
HOME=/Users/anttivainikka
ITERM_PROFILE=Default
ITERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
LC_CTYPE=UTF-8
LC_TERMINAL=iTerm2
LC_TERMINAL_VERSION=3.3.12
LOGNAME=anttivainikka
OMF_CONFIG=/Users/anttivainikka/.config/omf
OMF_PATH=/Users/anttivainikka/.local/share/omf
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
PWD=/Users/anttivainikka
SHELL=/usr/local/bin/fish
SHLVL=1
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.QY4EnwoKcA/Listeners
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.3.12
TERM_SESSION_ID=w0t0p0:3E6BD11C-8853-4A58-82AC-A8EA2429D0E1
TMPDIR=/var/folders/s6/16dyg8sd01d3b1xp4qwrx_h80000gn/T/
USER=anttivainikka
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
```

**Comment by jorgebucaran** [comment]
This is puzzling. Could you try uninstalling all your packages and just trying `fisher add franciscolourenco/done`?

**Comment by Vaintti** [comment]
That did the trick. And reinstalling the other packages didn't even break it. Really strange. Could it have been something that MacOS update could have broken? Like if an update did something to the file permissions

**Comment by Vaintti** [comment]
Rebooting and running fisher broke it again. Let me test with only franciscolourenco/done

**Comment by Vaintti** [comment]
Installing fishpkg/fish-prompt-metro  removed the dependency for some reason

**Comment by Vaintti** [comment]
It's really inconsistent. Sometimes it removes the humanize-duration dependency by installing something new or updating packages and sometimes it doesn't

**Comment by jorgebucaran** [comment]
Thank you, @Vaintti, I can confirm there's a bug. Now, I just need to figure out how to fix it!

**Comment by jorgebucaran** [comment]
I'm removing dependencies in Fisher 4.0. Also, `franciscolourenco/done` doesn't depend on `humanize_duration` anymore.

When I first introduced dependencies back in the day, I thought I'd use this feature all the time. Perhaps I was influenced by my background with Node.js The reality is that none of my plugins use dependencies now. I realized that dependencies don't make sense in a global context, since there is no way for a plugin to depend on another without interfering with the global scope.

No plugins/prompts listed on https://git.io/awesome.fish use dependencies either. If a plugin depends on another, it should be mentioned in the installation instructions.
